### ISSUE TITLE
BUG: Cannot build SciPy on x86 Mac due to gfortran lib location problem

### ISSUE THREAD

**Issue by AlexSzatmary** [scipy/scipy#18851]
### Describe your issue.

I am trying to build SciPy from source on a x86_64 MacOS 12.6.3 using these instructions:
https://docs.scipy.org/doc/scipy-1.9.1/dev/contributor/osx.html
I installed gfortran from the att.com site.

The issue is apparently that the SciPy builder only looks for libgfortran.5.dylib in /usr/lib/. I tried `export FLIBS="/usr/local/gfortran/lib"` but that didn't make a difference. That makes sense because when I searched the source code, I couldn't find the location gfortran being settable by an environment variable.

### Reproducing Code Example

```python
python dev.py test
```

### Error message

```shell
I get this error message (repeated, here's a typical example):

    from . import _arpack
E   ImportError: dlopen(/Users/alexszatmary/Dropbox/code/scipy/build-install/li
b/python3.11/site-packages/scipy/sparse/linalg/_eigen/arpack/_arpack.cpython-31
1-darwin.so, 0x0002): Library not loaded: '@rpath/libgfortran.5.dylib'
E     Referenced from: '/Users/alexszatmary/Dropbox/code/scipy/build-install/li
b/python3.11/site-packages/scipy/sparse/linalg/_eigen/arpack/_arpack.cpython-31
1-darwin.so'
E     Reason: tried: '/usr/lib/libgfortran.5.dylib' (no such file)
```
```


### SciPy/NumPy/Python version and system information

```shell
SciPy version not shown:

import sys, numpy; print(numpy.__version__, sys.version_info)            
1.25.1 sys.version_info(major=3, minor=11, micro=4, releaselevel='final', seria
l=0)
```

**Comment by andyfaff** [comment]
What version of scipy are you trying to build? The most recent release? from main? More recent versions have instructions that are easier to follow.

When I build and use from source (when I'm doing development work) I sometimes come across something similar. I think I set one of DYLD_LIBRARY_PATH, DYLD_FALLBACK_LIBRARY_PATH, PATH, but I can't remember which off the top of my head.

**Comment by AlexSzatmary** [comment]
I'm building from main. I'm wanting to become a contributor. I found the dev version of the build from source guide—thanks for the tip. `export DYLD_LIBRARY_PATH="/usr/local/gfortran/lib"` fixed things! Thanks!

So should anything change? I'm not familiar with the build tools but I think something should look in more reasonable locations on modern MacOS; in recent versions, I think it's not possible for users to install to just /usr. The build tools could at least check for the environment variable and act on it if set. The guide recommends installing gfortran using brew—I don't think that would solve the problem as gfortran would wind up in /opt/homebrew/Cellar.

**Comment by tupui** [comment]
On modern macOS and non intel, there should be no issues following our most recent instructions and recommendations: use a conda env and use dev.py.

Looks like we can close the issue. Thanks all

**Comment by andyfaff** [comment]
> So should anything change? I'm not familiar with the build tools but I think something should look in more reasonable locations on modern MacOS; 

Location of dylibs is only an issue on a development machine. It shouldn't be an issue when installing a wheel, because we "delocate" all the library locations when repairing the wheel.

On a dev machine it shouldn't be an issue when using dev.py to run tests. It may only bite you when you add the build directory to PYTHONPATH then start up an interpreter.

**Comment by andyfaff** [comment]
> On modern macOS and non intel, there should be no issues following our most recent instructions and recommendations: use a conda env and use dev.py.

I'm not sure that's the case. I don't have issues when using dev.py to run tests. However, if I add `build-install` to PYTHONPATH and try to use scipy from an interpreter I get the OP messages, which I compensate for by using DYLD_LIBRARY_PATH.

**Comment by tupui** [comment]
Yes but that's not a recommended workflow that we describe in our guides.

**Comment by andyfaff** [comment]
This kind of workflow is required

**Comment by tupui** [comment]
I am not sure what you are doing, I always do the following without any issue (conda, dev.py):

```
export PYTHONPATH=.../build-install/lib/python3.11/site-packages:
```

Also using `python dev.py shell/ipython` should do the same.

**Comment by AlexSzatmary** [comment]
I got things working by setting the environment variable so I'm set. However, it may still be worth making some kind of change. I went back and made sure I followed the exact steps in the dev version of the docs:
https://scipy.github.io/devdocs/building/index.html#building-from-source
I'm using the Apple developer tools and the python.org installer as before. I uninstalled gfortran from before and instead used homebrew to install it, as recommended in that guide. I have the same issue as before. Setting DYLD_LIBRARY_PATH (to the location that homebrew puts libgfortran.5.dylib) resolves the issue.
